e2e/nomostest/wait_for_sync.go

```diff
@@ -37,13 +39,12 @@ import (
 )
 
 type watchForAllSyncsOptions struct {
-	timeout            time.Duration
 	readyCheck         bool
 	syncRootRepos      bool
 	syncNamespaceRepos bool
 	skipRootRepos      map[string]bool
 	skipNonRootRepos   map[types.NamespacedName]bool
-	predicates         []testpredicates.Predicate
+	watchForSyncOpts   []WatchForSyncOption
```

Comment:
Maybe pull out this Predicates -> WatchForSyncOptions change to its own PR first to reduce the size of this one a bit.

Reply:
https://github.com/GoogleContainerTools/kpt-config-sync/pull/1594